src/battle_anim_new.c

```diff
@@ -4948,7 +7552,193 @@ static void SpriteCB_Geyser(struct Sprite* sprite)
+{
+	3, 3, 3, 3,
+	3, 3, 2, 2,
+	1, 1, 1, 1,
+	1, 1, 1, 1,
+	0, 1, 0, 1,
+	0, 1, 0, 0,
+	1, 0, 0, 1,
+	0, 0, 0, 1,
+	0, 0, 0, 1,
+};
+
+static const s8 sHomerunEnemyVerticalMovement[] =
+{
+	-4, -4, -4, -4,
+	-4, -3, -3, -2,
+	-2, -1, -1, -1,
+	-1, -1, -1, -1,
+	 0, -1,  0, -1,
+	 0, -1,  0,  0,
+	 0,  0, -1,  0,
+	 0, -1,  0,  0,
+	-1,  0,  0,  0,
+};
+
+void AnimTask_TwinkleTackleLaunchStep(u8 taskId)
+{
+	u16 rotation;
+	s16 xScale, yScale;
+	struct Task* task = &gTasks[taskId];
+	struct Sprite* sprite = &gSprites[task->tSpriteId];
+
+	if (task->tTimer > task->tAnimLengthTime)
+	{
+		if (task->tTimer > task->tAnimLengthTime + 5) //Wait an extra few frames so the glint can be placed on the target
+		{
+			sprite->pos1.x = task->tInitialXPos;
+			sprite->pos1.y = task->tInitialYPos;
+			ResetSpriteRotScale(task->tSpriteId);
+			DestroyAnimVisualTask(taskId);
+		}
+		else
+			++task->tTimer;
+		return;
+	}
+	else if ((u16) task->tTimer < NELEMS(sHomerunEnemyHorizontalMovement))
+	{
+		s8 movement = sHomerunEnemyHorizontalMovement[task->tTimer];
+		if (task->tSide == B_SIDE_PLAYER)
+			movement *= -1;
+		sprite->pos1.x += movement;
+
+		movement = sHomerunEnemyVerticalMovement[task->tTimer];
+		if (task->tSide == B_SIDE_PLAYER)
+			movement *= -1;
+		sprite->pos1.y += movement;
+	}
+
+	xScale = 0x180;
+	yScale = 0x180;
+	rotation = (task->tTimer << 4) + (task->tTimer << 3);
+
+	xScale += rotation;
+	yScale += rotation;
+	rotation <<= 7;
+
+	if (task->tSide == B_SIDE_OPPONENT)
+		rotation *= -1;
+
+	SetSpriteRotScale(task->tSpriteId, xScale, yScale, rotation);
+
+	if (++task->tTimer > task->tAnimLengthTime)
+		sprite->invisible = TRUE;
+}
+
+//Launches the target in Twinkle Tackle
+//arg 0: Anim time
+void AnimTask_TwinkleTackleLaunch(u8 taskId)
+{
+	struct Task* task = &gTasks[taskId];
+
+	task->tSpriteId = GetAnimBattlerSpriteId(ANIM_TARGET);
+	task->tSide = GetBattlerSide(gBattleAnimTarget);
+	task->tAnimLengthTime = gBattleAnimArgs[0];
+	task->tInitialXPos = gSprites[task->tSpriteId].pos1.x;
+	task->tInitialYPos = gSprites[task->tSpriteId].pos1.y;
+	task->tTimer = 0;
+	task->func = AnimTask_TwinkleTackleLaunchStep;
+
+	PrepareBattlerSpriteForRotScale(task->tSpriteId, ST_OAM_OBJ_NORMAL);
+}
+#undef tSpriteId
+#undef tTimer
+#undef tInitialXPos
+#undef tInitialYPos
+#undef tSide
+#undef tAnimLengthTime
+
+void AnimTask_GetTimeOfDay(u8 taskId)
+{
+	gBattleAnimArgs[0] = 0; //Daytime is default
+
+    RtcCalcLocalTime();
+    if (gLocalTime.hours >= 20 || gLocalTime.hours < 4)
+        gBattleAnimArgs[0] = 1;
+    else if (gLocalTime.hours >= 17 && gLocalTime.hours < 20)
+        gBattleAnimArgs[0] = 2;
+    
+	DestroyAnimVisualTask(taskId);
+}
+
+void AnimTask_GetLycanrocForm(u8 taskId)
+{
+	/* TODO - lycanroc forms
+    #ifdef SPECIES_LYCANROC_MIDNIGHT
+	if (GetIllusionPartyData(gBattleAnimAttacker)->species == SPECIES_LYCANROC_MIDNIGHT)
+		gBattleAnimArgs[0] = 1;
+	else
+	#endif
+		gBattleAnimArgs[0] = 0;*/
```

Comment:
is there a reason why this is todo? i'm pretty sure PE has midnight lycanroc so there shouldn't be an issue with this check